### PR TITLE
feat(indentscope): show indent guide on wrapped lines

### LINKED ISSUE #635: feat: continue indent guides on wrapped lines
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

mini.indentscope

## What
Nightly now includes the ability to [display extmarks on wrapped lines](https://github.com/neovim/neovim/issues/23108). This feature has made its way into other indent guide plugins (see [here](https://github.com/lukas-reineke/indent-blankline.nvim/pull/810) and [here](https://github.com/lukas-reineke/indent-blankline.nvim/pull/812)). 

## How
Using nightly, seeing the effects of this feature only requires placing `virt_text_repeat_linebreak = true` into the `extmark_opts` table in indentscope's `make_draw_function()`. In the below example, I am using indentscope with this option on, and it is visible on the wrapped lines indicated by dashes in the number columns.

break-with-guide

The bulk of the change seems like it would be adding error handling and checking the user's Neovim version (until 0.10 is released), options, and indentscope configuration.

### PR DESCRIPTION
**Pull request by gpanders**
Resolve #635.

- [x] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [x] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: tests/test_indentscope.lua [hunk 1]
```diff
@@ -509,6 +509,21 @@ T['draw()']['does not round time of every animation step'] = function()
   child.expect_screenshot()
 end
 
+T['draw()']['shows symbols on wrapped lines'] = function()
+  -- Add a long line of text. Set 'breakindent' and 'breakindentopt' to ensure
+  -- the wrapped line has the same indent as the current line
+  set_cursor(6, 1)
+  child.cmd('normal! 10Aa')
```
Comment (echasnovski) [inline tests/test_indentscope.lua:516]: Maybe `set_lines()` with some smaller set of lines together with explicit `child.set_size()` makes the test clearer?

**Review by echasnovski — CHANGES_REQUESTED** [review]
Thanks so much for this!

Overall it looks good, only stylistic nit picks.

Also would you mind using "Resolve #XXX" in commit message (instead of "Resolve echasnovski#XXX")?

File: tests/test_indentscope.lua (hunk 1 above)
Comment (gpanders) [reply tests/test_indentscope.lua]: b55794d6f958faf4d101bfb0cefb3943cc794311

File: tests/test_indentscope.lua [hunk 2]
```diff
@@ -509,6 +509,21 @@ T['draw()']['does not round time of every animation step'] = function()
   child.expect_screenshot()
 end
 
+T['draw()']['shows symbols on wrapped lines'] = function()
```
[the remark this answers is not in the document]
Comment (gpanders) [reply tests/test_indentscope.lua:512]: bf97cdda4f9ee2bb6bb426e0497a977cbde0f13a

File: lua/mini/indentscope.lua [hunk 3]
```diff
@@ -938,6 +938,8 @@ H.make_draw_function = function(indicator, opts)
     virt_text_pos = 'overlay',
   }
 
+  if vim.fn.has('nvim-0.10') == 1 then extmark_opts.virt_text_repeat_linebreak = true end
```
[the remark this answers is not in the document]
Comment (gpanders) [reply lua/mini/indentscope.lua]: 62104be7c288f0f62c247caac4167b04fb31460e

**Review by echasnovski — APPROVED** [review]
Nice! Thanks for the follow up!

I'll merge in temporary branch and clean up the loose ends.

**Pull request merged**